DEVELOPER_GUIDE.rst

```diff
@@ -182,17 +182,23 @@ Note that other related project code has already merged into this single reposit
 - ``workbench``: query workbench UI.
 
 
-Code Convention
+Code Convention (Spotless)
 ---------------
 
-We’re integrated Checkstyle plugin into Gradle build: https://github.com/opensearch-project/sql/blob/main/config/checkstyle/google_checks.xml. So any violation will fail the build. You need to identify the offending code from Gradle error message and fix them and rerun the Gradle build. Here are the highlight of some Checkstyle rules:
-
-* 2 spaces indentation.
-* No line starts with tab character in source file.
-* Line width <= 100 characters.
-* Wildcard imports: You can enforce single import by configuring your IDE. Instructions for Intellij IDEA: https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcard-imports.
-* Operator needs to wrap at next line.
-
+Java files are formatted using `Spotless <https://github.com/diffplug/spotless>` conforming to `Google Java Format <https://github.com/google/google-java-format>`.
```

Comment:
https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages
needs `_`

Reply:
Thanks, addressed

Reply:
Let's create an opening paragraph like https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#java-language-formatting-guidelines

How about: 
```
Java files in the OpenSearch codebase are formatted with the Eclipse JDT formatter, using the [Spotless Gradle](https://github.com/diffplug/spotless/tree/master/plugin-gradle) plugin. This plugin is configured in the project  `./gradle.properties`. 

The formatting check can be run explicitly with:
./gradlew spotlessJavaCheck

The code can be formatted with: 
./gradlew spotlessApply

These tasks can also be run for specific modules, e.g.
./gradlew server:spotlessJavaCheck

For more information on the spotless for the OpenSearch project please see `https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#java-language-formatting-guidelines`_.
```

Reply:
Note: the only difference between what core uses and us is the gradle properties... which the core documentation stays should be configured on a project-by-project basis anyways.

Reply:
Updated